src/util/side_metadata/side_metadata_tests.rs

```diff
@@ -221,6 +221,8 @@ mod tests {
             lspec.log_min_obj_size = 4;
             lspec.log_num_of_bits = 2;
 
+            sanity::reset();
```

Comment:
If any assertion fails, the `reset` won't be called. If the reset is a cleanup procedure, you can use `with_cleanup()` from `test_util`.

Reply:
The sanity code uses static maps which are global. So even though these few tests are `serial_test`, the sanity maps may get mutated by other tests.

Reply:
I changed the relevant tests to use `with_cleanup`.

As for the second point, we need to make all tests accessing sanity serial.
I don't think we have any issues with this now.

Reply:
Is it true that all the tests for side metadata are using serial_test? Any test that uses side metadata may modify the maps with extreme_assertions turned on.

Reply:
I went through all tests and found that the ones with static instances of SideMetadata (e.g. `NO_METADATA`) could cause issues because we're not stateless with regards to no metadata anymore and `NO_METADATA` should belong to a test and be cleaned up after the test is finished.
So, I changed all static instances to per-test instance and added `sanity::zero` to the cleanup for all metadata related tests.

Reply:
What if we write tests for a space? A space would use side metadata, and modifies the static map.

Reply:
Such tests should be:
1 - serial,
2 - use exclusive metadata instances,
3 - use `sanity::reset` in their cleanup

I couldn't find any tests in this category though

Reply:
Okay. Can you add the above as comments somewhere? The idea is if someone trips when writing tests, they shhould be able to find the comments to tell them what to do.

Reply:
Done